File: projects/2/michael_ezeigbo/hello.py
import sys

def print_greeting(name, count):

    print(f"Starting to greet {name}...")

    for i in range(count):
        print(f"Hello {name}!")

    print(f"Greeting process completed for {name}.")

def main():

    if len(sys.argv) != 3:
        print("Usage: python hello.py <name> <number>")
        sys.exit(1)

    name = sys.argv[1]
    try:
        count = int(sys.argv[2])
    except ValueError:
        print("Error: The number of greetings must be an integer.")
        sys.exit(1)

  
    if count <= 0:
        print("Error: The number of greetings must be a positive integer.")
        sys.exit(1)


    print_greeting(name, count)

if __name__ == "__main__":
    main()

File: projects/2/michael_ezeigbo/hello.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	
	if len(os.Args) != 3 {
		fmt.Println("Usage: ./hello <name> <number>")
		return
	}

	
	name := os.Args[1]
	count, err := strconv.Atoi(os.Args[2])
	if err != nil {
		fmt.Println("Error: The number of greetings must be an integer.")
		return
	}


	if count <= 0 {
		fmt.Println("Error: The number of greetings must be a positive integer.")
		return
	}

	
	fmt.Printf("Starting to greet %s...\n", name)
	for i := 0; i < count; i++ {
		fmt.Printf("Hello %s!\n", name)
	}
	fmt.Printf("Greeting process completed for %s.\n", name)
